### PR TITLE
Feature/fate#326330

### PR DESCRIPTION
**Pull request by lproven**
This is the complete 1st draft of the new section on NVDIMM support. Comments sought!

### CODE REVIEW

**Comment by cwickert** [comment]
First impression: Looks good, but I'm not happy with putting this under a new part with just a single chapter. Not sure if there is a better place.

Stay tuned for a full review.

**Comment by lproven** [comment]
> First impression: Looks good

Glad to hear it! I have tried to keep my XML cleaner, better-formatted, make fewer, bigger commits, and generally learn from prior feedback.

> but I'm not happy with putting this under a new part with just a single chapter. Not sure if there is a better place.

This was @svenseeberg's suggestion and is based on the Hardware § of the SLE 15 Admin Guide:

https://susedoc.github.io/doc-sle/develop/SLES-admin/single-html/#part.hardware

I took the source from there and replaced his VM chapter with my NVDIMM chapter. My hope was that this would make future integration easier.

I do not know the decision-making behind the was it was done for SLE 15.

**Comment by lproven** [comment]
> First impression: Looks good

Glad to hear it! I have tried to keep my XML cleaner, better-formatted, make fewer, bigger commits, and generally learn from prior feedback.

> but I'm not happy with putting this under a new part with just a single chapter. Not sure if there is a better place.

This was @svenseeberg's suggestion and is based on the Hardware § of the SLE 15 Admin Guide:

https://susedoc.github.io/doc-sle/develop/SLES-admin/single-html/#part.hardware

I took the source from there and replaced his VM chapter with my NVDIMM chapter. My hope was that this would make future integration easier.

I do not know the decision-making behind the way it was done for SLE 15.

File: xml/hardware_nvdimm.xml [hunk 1]
```diff
@@ -0,0 +1,887 @@
...
+     </variablelist>
+     <para>
+      Some devices support both PMEM and BLK modes. Additionally, some allow
+      the storage to be split into separate namespaces, so that some can be
+      accessed using PMEM and some using BLK.
+     </para>
+     <para>
+      Apart from <literal>devdax</literal> namespaces, all other types must be
+      formatted with a filesystem, just as with a conventional drive.
+      &productname; supports the <literal>ext2</literal>, <literal>ext4</literal> and
```
Comment (cwickert) [inline xml/hardware_nvdimm.xml]: line too long

File: xml/hardware_nvdimm.xml [hunk 2]
```diff
@@ -0,0 +1,887 @@
...
+    </title> 
+    <para>
+     In cases where the server will be hosting an application that can directly
+     access large amounts of storage on a byte-by-byte basis, by 
+     <literal>mmap</literal>ing the storage directly into the application's
+     address space.
+    </para>
+   </sect3>
+   <sect3>
+    <title>
```
Comment (cwickert) [inline xml/hardware_nvdimm.xml]: no newlines

File: xml/hardware_nvdimm.xml [hunk 3]
```diff
@@ -0,0 +1,887 @@
...
+  </sect2>
+  <sect2>
+   <title>BLK storage</title>
+   <para>
+    Although it is more robust against single-device failure, this requires
+    additional management, as each NVDIMM appears as a separate device. Thus,
+    PMEM is generally preferred.
+   </para>
+   <para>
+    BLK storage is deprecated and is not supported in later versions of &productname;.
```
Comment (cwickert) [inline xml/hardware_nvdimm.xml]: line too long

File: xml/hardware_nvdimm.xml [hunk 4]
```diff
@@ -0,0 +1,887 @@
...
+    </para>
+   </listitem>
+  </itemizedlist>
+
+  <para>
+   The <command>ndctl</command> utility has a helpful set of
+   <command>man</command> pages, accessible with the command:
+  </para>
+
+<screen>ndctl help <replaceable>subcommand</replaceable></screen>
```
Comment (cwickert) [inline xml/hardware_nvdimm.xml]: command

File: xml/hardware_nvdimm.xml [hunk 5]
```diff
@@ -0,0 +1,887 @@
...
+
+  <para>
+   To see a list of available subcommands, use:
+  </para>
+
+<screen>ndctl --list-cmds</screen>
+
+  <para>
+   The available subcommands include:
```
Comment (cwickert) [inline xml/hardware_nvdimm.xml]: command

File: xml/hardware_nvdimm.xml [hunk 6]
```diff
@@ -0,0 +1,887 @@
...
+   <title>Viewing Available NVDIMM Storage</title>
+   <para>
+    The <command>ndctl</command> <literal>list</literal> command can be used to
+    list all available NVDIMMs in a system.
+   </para>
+   <para>
+    In the following example, the system has three NVDIMMs which are in a
+    single, triple-channel interleaved set.
+   </para>
+<screen>&prompt.root;ndctl list --dimms
```
Comment (cwickert) [inline xml/hardware_nvdimm.xml]: command

File: xml/hardware_nvdimm.xml [hunk 7]
```diff
@@ -0,0 +1,887 @@
...
+   <note>
+    <para>
+     Regions may not appear in numerical order.
+    </para>
+   </note>
+   <para>
+    Note that although there are only three NVDIMMs, they appear as four
+    regions.
+   </para>
+<screen>&prompt.root;ndctl list --regions
```
Comment (cwickert) [inline xml/hardware_nvdimm.xml]: command

File: xml/hardware_nvdimm.xml [hunk 8]
```diff
@@ -0,0 +1,887 @@
...
+  <sect2 xml:id="sec.nvdimm.setup.dax">
+   <title>Configuring the Storage as a Single PMEM Namespace with DAX</title>
+   <para>
+    For the first example, we will configure our three NVDIMMs into a single
+    PMEM namespace with Direct Access (DAX).
+   </para>
+   <para>
+    The first step is to create a new namespace.
+   </para>
+<screen>&prompt.root;ndctl create-namespace --type=pmem --mode=memory
```
Comment (cwickert) [inline xml/hardware_nvdimm.xml]: command

File: xml/hardware_nvdimm.xml [hunk 9]
```diff
@@ -0,0 +1,887 @@
...
+   <para>
+    The reservation of some persistent memory for kernel data structures is why
+    the resulting PMEM namespace has a smaller capacity than the parent PMEM
+    Region.
+   </para>
+   <para>
+    Next, we verify that the new block device is available to the operating
+    system:
+   </para>
+<screen>&prompt.root;fdisk -l /dev/pmem3
```
Comment (cwickert) [inline xml/hardware_nvdimm.xml]: command

File: xml/hardware_nvdimm.xml [hunk 10]
```diff
@@ -0,0 +1,887 @@
...
+Disk /dev/pmem3: 186 GiB, 199764213760 bytes, 390164480 sectors
+Units: sectors of 1 * 512 = 512 bytes
+Sector size (logical/physical): 512 bytes / 4096 bytes
+I/O size (minimum/optimal): 4096 bytes / 4096 bytes</screen>
+   <para>
+    Before it can be used, like any other drive, it must be formatted. In this
+    example, we format it with XFS:
+   </para>
+<screen>&prompt.root;mkfs.xfs /dev/pmem3
```
Comment (cwickert) [inline xml/hardware_nvdimm.xml]: command, replacable

**Comment by cwickert** [comment]
I fixed the remaining bits myself, nothing really serious.

**Pull request merged**